File: PHP/7php 1/binaryArrayToNumber.php
<?php
/*
|--------------------------------------------------------------------------
| Ones and Zeros
|--------------------------------------------------------------------------
|
*/

function binaryArrayToNumber($arr) {
   return bindec(implode('', $arr));
}


function binaryArrayToNumber($arr) {
   $final = 0;
   for ($i = 0; $i < count($arr); $i++) {
      $final += ($arr[count($arr)-($i+1)] * (2**$i));
   }
   return $final;
}


function binaryArrayToNumber($arr) {
   return base_convert(implode($arr),2,10);
}


function binaryArrayToNumber($arr) {
   $num = 0;
   foreach($arr as $digit){
      $num *= 2;
      $num += $digit == 1;
   }
   return $num;
}


function binaryArrayToNumber(array $arr): int {
   return array_reduce($arr, function ($result, $digit) {
      return $result * 2 + $digit;
   }, 0);
}


print_r(binaryArrayToNumber([0, 0, 0, 1]). ' ' . 1 . '
');
print_r(binaryArrayToNumber([0, 0, 1, 0]). ' ' . 2 . '
');
print_r(binaryArrayToNumber([1, 1, 1, 1]). ' ' . 15 . '
');
print_r(binaryArrayToNumber([0, 1, 1, 0]). ' ' . 6 . '
');

File: Python/7py 1/binary_array_to_number.py
"""
|--------------------------------------------------------------------------
| Ones and Zeros
|--------------------------------------------------------------------------
|
"""

def binary_array_to_number(arr):
   return int(''.join(list(map(lambda i: str(i), arr))), 2)


def binary_array_to_number(arr):
   return int("".join(map(str, arr)), 2)


def binary_array_to_number(arr):
   return int(''.join(str(a) for a in arr), 2)


def binary_array_to_number(arr):
   return sum(digit * 2**i for i, digit in enumerate(reversed(arr)))


print(binary_array_to_number([0,0,0,1]), 1)
print(binary_array_to_number([0,0,1,0]), 2)
print(binary_array_to_number([1,1,1,1]), 15)
print(binary_array_to_number([0,1,1,0]), 6)